### PR TITLE
ddtrace/tracer: updated DD_TAGS parsing to support space separation

### PR DESCRIPTION
**Pull request by dianashevchenko**
DD_TAGS were split by commas only. Unless there is a comma in a string, tag list will be split by spaces. In either case, commas/spaces will be trimmed.

### CODE REVIEW

File: ddtrace/tracer/option_test.go [hunk 1]
```diff
@@ -249,6 +249,84 @@ func TestServiceName(t *testing.T) {
...
+					"env":  "test",
+					"bKey": "",
+					"dKey": "",
+					"dVal": "",
+					"cKey": ""}},
+			{
+				in: "env :test, aKey : aVal bKey:bVal cKey:",
+				expected: map[string]string{
+					"env":  "test",
+					"aKey": "aVal bKey:bVal cKey:"}},
```
Comment (gbbr) [inline ddtrace/tracer/option_test.go:291]: The last curly should go on the next line.

File: ddtrace/tracer/option.go [hunk 2]
```diff
@@ -138,19 +138,25 @@ func newConfig(opts ...StartOption) *config {
 		c.version = ver
 	}
 	if v := os.Getenv("DD_TAGS"); v != "" {
-		for _, tag := range strings.Split(v, ",") {
-			tag = strings.TrimSpace(tag)
+		sep, trim := " ", ","
```
Comment (gbbr) [inline ddtrace/tracer/option.go:141]:
```suggestion
		sep  := " "
```
Let's drop that. I don't see any reason to trim commas, do you?

File: ddtrace/tracer/option.go [hunk 3]
```diff
@@ -138,19 +138,25 @@ func newConfig(opts ...StartOption) *config {
...
+		if strings.Index(v, ",") > -1 {
+			sep, trim = trim, sep
+		}
+		for _, tag := range strings.Split(v, sep) {
+			tag = strings.Trim(tag, trim)
```
Comment (gbbr) [inline ddtrace/tracer/option.go:146]:
```suggestion
			tag = strings.TrimSpace(tag)
```
Should be good enough in all cases, even if it's redundant when we choose space as a separator.

File: ddtrace/tracer/option.go [hunk 4]
```diff
@@ -138,19 +138,25 @@ func newConfig(opts ...StartOption) *config {
...
 				continue
 			}
 			kv := strings.SplitN(tag, ":", 2)
-			k := strings.TrimSpace(kv[0])
-			switch len(kv) {
-			case 1:
-				WithGlobalTag(k, "")(c)
-			case 2:
-				WithGlobalTag(k, strings.TrimSpace(kv[1]))(c)
+			key := strings.Trim(kv[0], trim)
```
Comment (gbbr) [inline ddtrace/tracer/option.go:151]: Same here and elsewhere.

Comment (dianashevchenko) [reply ddtrace/tracer/option.go:151]: There was no switch clause left, what do you reference here?

Comment (gbbr) [reply ddtrace/tracer/option.go:151]: I mean to use `TrimSpace`

File: ddtrace/tracer/option.go [hunk 5]
```diff
@@ -138,19 +138,25 @@ func newConfig(opts ...StartOption) *config {
 		c.version = ver
 	}
 	if v := os.Getenv("DD_TAGS"); v != "" {
-		for _, tag := range strings.Split(v, ",") {
+		sep := " "
+		if strings.Index(v, ",") > -1 {
+			sep = ","
```
Comment (gbbr) [inline ddtrace/tracer/option.go:143]: Let's add a comment above this line
```suggestion
			// falling back to comma as separator
			sep = ","
```

File: ddtrace/tracer/option_test.go [hunk 6]
```diff
@@ -249,6 +249,94 @@ func TestServiceName(t *testing.T) {
...
+				expected: map[string]string{
+					"a": "b",
+					"c": "",
+					"d": ""},
+			},
+			{
+				in: "a,1",
+				expected: map[string]string{
+					"a": "",
+					"1": ""},
```
Comment (gbbr) [inline ddtrace/tracer/option_test.go:321]: This curly brace should also go on the next line everywhere. Sorry if my initial message was unclear.

File: ddtrace/tracer/option_test.go [hunk 7]
```diff
@@ -249,6 +249,94 @@ func TestServiceName(t *testing.T) {
...
+			},
+			{
+				in:       "a:b:c:d",
+				expected: map[string]string{"a": "b:c:d"},
+			},
+		} {
+			os.Setenv("DD_TAGS", tag.in)
+			c := newConfig()
+			for key, expected := range tag.expected {
+				actual, ok := c.globalTags[key]
```
Comment (gbbr) [inline ddtrace/tracer/option_test.go:331]:
```suggestion
				v, ok := c.globalTags[key]
```
Can we please use `v` or `got`?

File: ddtrace/tracer/option_test.go [hunk 8]
```diff
@@ -249,6 +249,94 @@ func TestServiceName(t *testing.T) {
...
+				assert.Truef(ok, "ind %v, exp %v", ind, expected)
```
Comment (gbbr) [inline ddtrace/tracer/option_test.go:332]: Just do this
```suggestion
				assert.True(ok, "tag not found")
```

File: ddtrace/tracer/option_test.go [hunk 9]
```diff
@@ -249,6 +249,94 @@ func TestServiceName(t *testing.T) {
...
+				assert.Equal(expected, actual)
+			}
+		}
+		defer os.Unsetenv("DD_TAGS")
```
Comment (gbbr) [inline ddtrace/tracer/option_test.go:336]: `defer` is usually at the top 🙂 Furthermore, if the test fails anywhere, this call will never happen, defeating the purpose of deferring.

File: ddtrace/tracer/option_test.go [hunk 10]
```diff
@@ -249,6 +249,94 @@ func TestServiceName(t *testing.T) {
...
+				expected: map[string]string{
+					"a": "",
+					"1": ""},
+			},
+			{
+				in:       "a:b:c:d",
+				expected: map[string]string{"a": "b:c:d"},
+			},
+		} {
+			os.Setenv("DD_TAGS", tag.in)
```
Comment (gbbr) [inline ddtrace/tracer/option_test.go:328]:
```suggestion
			t.Run("", func(t *testing.T) {
				os.Setenv("DD_TAGS", tag.in)
```
Tip: if you do this, with empty string sub-test names, Go will automatically count each test and you'll know which one failed. With this method, you can drop `ind` (which should generally be `i` when used)

File: ddtrace/tracer/option_test.go [hunk 11]
```diff
@@ -249,6 +249,94 @@ func TestServiceName(t *testing.T) {
 	})
 }
 
+func TestTagSeparators(t *testing.T) {
+	assert := assert.New(t)
+
+	t.Run("env-tags", func(t *testing.T) {
+		for ind, tag := range []struct {
+			in       string
+			expected map[string]string
```
Comment (gbbr) [inline ddtrace/tracer/option_test.go:258]:
```suggestion
			out map[string]string
```
Best friend of `in` is `out` 🙂

File: profiler/options.go [hunk 12]
```diff
@@ -156,8 +156,12 @@ func defaultConfig() *config {
 		WithVersion(v)(&c)
 	}
 	if v := os.Getenv("DD_TAGS"); v != "" {
-		for _, tag := range strings.Split(v, ",") {
-			tag = strings.TrimSpace(tag)
+		sep, trim := " ", ","
```
Comment (gbbr) [inline profiler/options.go:159]: Please make the same changes here.

File: ddtrace/tracer/option_test.go (hunk 7 above)
Comment (dianashevchenko) [reply ddtrace/tracer/option_test.go:331]: v is used everywhere above, so I did not change that

File: ddtrace/tracer/option_test.go (hunk 9 above)
Comment (dianashevchenko) [reply ddtrace/tracer/option_test.go:336]: Interestingly enough, the tests above are not different. I'll delete that line then?

Comment (gbbr) [reply ddtrace/tracer/option_test.go]: I can't see any test above where `defer` is called at the end of the function.

File: ddtrace/tracer/option_test.go (hunk 7 above)
Comment (gbbr) [reply ddtrace/tracer/option_test.go:339]: What do you mean?

**Review by gbbr — APPROVED** [review]
Looks perfect to me. CI is not passing though, there's one check you missed....

**Pull request merged**